lib/loadObj.js

```diff
@@ -494,6 +514,23 @@ function getImagePaths(materials) {
     return Object.keys(imagePaths);
 }
 
+function assignDefaultMaterial(nodes, materials) {
+    var defaultMaterial = materials[0].name;
```

Comment:
Same as above, looks like `materials` is an object. Maybe this pathway needs a separate spec? I only managed to get here with an `obj` that didn't `usemtl` but had a material library.

Reply:
Maybe there's already a spec for this case but it's currently too permissive to catch the bug above + here.

Reply:
Thanks for catching this. I had brought over the functionality without bringing over the tests.

As I was looking at which tests needed to be brought over I noticed that a lot of PRs that made it into master never made it into `1.0`. This PR kind of blew up because of that and I renamed the title accordingly...

The only PR I didn't bring over is https://github.com/AnalyticalGraphicsInc/obj2gltf/pull/124 which is a lot more work than the others and isn't super common to see. The alpha texture would need to be written to the diffuse texture alpha channel which requires all the raw pixel reading/packing in `master`.

Reply:
I will also test all the obj models I have tomorrow to make sure nothing broke.

Reply:
Also some PRs were not needed to back port:
* https://github.com/AnalyticalGraphicsInc/obj2gltf/pull/108
* https://github.com/AnalyticalGraphicsInc/obj2gltf/pull/115
* https://github.com/AnalyticalGraphicsInc/obj2gltf/pull/126